### PR TITLE
Add one-liner to observe in Activity and Fragment

### PR DESCRIPTION
**Pull request by mattmook**

### CODE REVIEW

File: buildSrc/src/main/kotlin/DependencyManagement.kt [hunk 1]
```diff
@@ -32,14 +32,12 @@ object Versions {
     const val kotlin = "1.5.20"
     const val coroutines = "1.5.0-native-mt"
 
-    const val androidxLifecycles = "2.2.0"
+    const val androidxLifecycles = "2.4.0-alpha02"
```
Comment (Rosomack) [inline buildSrc/src/main/kotlin/DependencyManagement.kt:35]: Depending on an alpha version is naff and my only concern here

File: orbit-viewmodel/src/main/kotlin/org/orbitmvi/orbit/viewmodel/ContainerHostExtensions.kt [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    sideEffect: (suspend (sideEffect: SIDE_EFFECT) -> Unit)? = null
+) {
+    lifecycleOwner.lifecycleScope.launch {
+        // See https://medium.com/androiddevelopers/a-safer-way-to-collect-flows-from-android-uis-23080b1f8bda
+        lifecycleOwner.lifecycle.repeatOnLifecycle(Lifecycle.State.STARTED) {
+            state?.let { launch { container.stateFlow.collect(state) } }
+            sideEffect?.let { launch { container.sideEffectFlow.collect(sideEffect) } }
+        }
+    }
+}
```
Comment (Rosomack) [inline orbit-viewmodel/src/main/kotlin/org/orbitmvi/orbit/viewmodel/ContainerHostExtensions.kt:56]: What happens if this is called multiple times? E.g. someone shoves it into `onResume` ?

**Review by Rosomack — CHANGES_REQUESTED** [review]

**Review by Rosomack — COMMENTED** [review]
Just a few comments

File: orbit-viewmodel/src/main/kotlin/org/orbitmvi/orbit/viewmodel/ContainerHostExtensions.kt (hunk 2 above)
Comment (mattmook) [reply orbit-viewmodel/src/main/kotlin/org/orbitmvi/orbit/viewmodel/ContainerHostExtensions.kt:56]: bad stuff... in future version we can add a lint rule... I have the code for one already saved

File: buildSrc/src/main/kotlin/DependencyManagement.kt (hunk 1 above)
Comment (mattmook) [reply buildSrc/src/main/kotlin/DependencyManagement.kt:35]: yeah I agree

we have two options:
- we just keep the PR open until such a time it is released; with the knowledge there are issues with lifecycle and side effects
- we merge it in as is and make someone else sad

Comment (Rosomack) [reply buildSrc/src/main/kotlin/DependencyManagement.kt:35]: I think in this instance it's more important to fix lifecycle defects, let's merge.

File: orbit-viewmodel/src/main/kotlin/org/orbitmvi/orbit/viewmodel/ContainerHostExtensions.kt (hunk 2 above)
Comment (Rosomack) [reply orbit-viewmodel/src/main/kotlin/org/orbitmvi/orbit/viewmodel/ContainerHostExtensions.kt:56]: OK, it's easy to abuse so we should be extra careful with the advice we give - lint rule would be useful

**Review by Rosomack — APPROVED** [review]

**Pull request merged**